src/Core/Prefab/Globe/GlobeTileBuilder.ts

```diff
@@ -0,0 +1,170 @@
+const axisY = new THREE.Vector3(0, 1, 0);
+const quatToAlignLongitude = new THREE.Quaternion();
+const quatToAlignLatitude = new THREE.Quaternion();
+const quatNormalToZ = new THREE.Quaternion();
+
+function WGS84ToOneSubY(latitude: number) {
+    return 1.0 - (0.5 - Math.log(Math.tan(
+        PI_OV_FOUR + THREE.MathUtils.degToRad(latitude) * 0.5,
+    )) * INV_TWO_PI);
+}
+
+export interface GlobeTileBuilderParams extends TileBuilderParams {
+    nbRow: number;
+    deltaUV1: number;
+    quatNormalToZ: THREE.Quaternion;
+}
+
+export class GlobeTileBuilder
+implements TileBuilder<GlobeTileBuilderParams> {
+    private _crs: string;
+    private _transform: {
+        coords: Coordinates[];
+        position: THREE.Vector3;
+        dimension: THREE.Vector2;
+    };
+
+    public computeExtraOffset?: (params: GlobeTileBuilderParams) => number;
+
+    public get crs(): string {
+        return this._crs;
+    }
+
+    public constructor(options: {
+        crs: string,
+        uvCount: number,
+    }) {
+        this._transform = {
+            coords: [
+                new Coordinates('EPSG:4326', 0, 0),
+                new Coordinates('EPSG:4326', 0, 0),
+            ],
+            position: new THREE.Vector3(),
+            dimension: new THREE.Vector2(),
+        };
+
+        this._crs = options.crs;
+        // Order crs projection on tiles
+
+        // UV: Normalized coordinates (from degree) on the entire tile
+        // EPSG:4326
+        // Offset: Float row coordinate from Pseudo mercator coordinates
+        // EPSG:3857
+        if (options.uvCount > 1) {
+            this.computeExtraOffset = GlobeTileBuilder._computeExtraOffset;
+        }
+    }
+
+    private static _computeExtraOffset(params: GlobeTileBuilderParams): number {
+        const t = WGS84ToOneSubY(params.projected.latitude) * params.nbRow;
+        return (!isFinite(t) ? 0 : t) - params.deltaUV1;
+    }
+
+    // prepare params
+    // init projected object -> params.projected
+    public prepare(params: TileBuilderParams): GlobeTileBuilderParams {
+        const nbRow = 2 ** (params.level + 1.0);
+        let st1 = WGS84ToOneSubY(params.extent.south);
+
+        if (!isFinite(st1)) { st1 = 0; }
+
+        const sizeTexture = 1.0 / nbRow;
+
+        const start = (st1 % (sizeTexture));
+
+        const newParams = {
+            nbRow,
+            deltaUV1: (st1 - start) * nbRow,
+            // transformation to align tile's normal to z axis
+            quatNormalToZ: quatNormalToZ.setFromAxisAngle(
+                axisY,
+                -(Math.PI * 0.5 - THREE.MathUtils.degToRad(
+                    params.extent.center().latitude,
+                ))),
+            // let's avoid building too much temp objects
+            projected: new Projected(),
+        };
+
+        params.extent.planarDimensions(this._transform.dimension);
+
+        return { ...params, ...newParams };
+    }
+
+    // get center tile in cartesian 3D
```

Comment:
I'd argue this isn't even aesthetics, jsdoc has become the _de facto_ standard for JS/TS comments. Those comments were just left behind from because I forgot clean them up, oops.